### PR TITLE
Wip/hardboprobot/nordic

### PR DESCRIPTION
**Pull request by d4s**
Type of pull request:

- [x ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

- [ x] Fill out `README.md` with update protocol
- [ x] N/A: ~~Fill out `README.md` with any custom quirks and flags~~ 
- [ x] Fill out `README.md` with the vendor ID security value
- [ x] N/A: ~~Implement `FuFirmware->write()` and include at least one fuzzer testcase in `src/fuzzing/firmware` for any custom `FuFirmware` subclass~~
- [ ] CI run of the plugin for at least one target
- [ ] Document targets CI isn't currently run and the reasons (i.e. minimum versions needed or distribution limitations etc).

### CODE REVIEW

File: plugins/nordic-hid/README.md
Comment (d4s) [reply plugins/nordic-hid/README.md]: Banks for A/B, however the format of FW has been changed and nowadays the bank is not the part of GUID.

File: plugins/nordic-hid/fu-nordic-hid-cfg-channel.c
Comment (d4s) [reply plugins/nordic-hid/fu-nordic-hid-cfg-channel.c]: Ha! good catch... there is no any call where `check_status` is set to `FALSE`, so unneeded at all.
Thanks for noticing that!

File: plugins/nordic-hid/fu-nordic-hid-archive.c
Comment (hughsie) [inline plugins/nordic-hid/fu-nordic-hid-archive.c:26]: Unused, no?

File: plugins/nordic-hid/fu-nordic-hid-archive.c
Comment (hughsie) [inline plugins/nordic-hid/fu-nordic-hid-archive.c:30]: unused?

File: plugins/nordic-hid/fu-nordic-hid-cfg-channel.c
Comment (hughsie) [inline plugins/nordic-hid/fu-nordic-hid-cfg-channel.c:979]: This looks fishy. Maybe get the checksum as a `const gchar *csum` and then return FALSE if that fails. Then check for checksum=0 and do a `g_set_error()` in that case.

File: plugins/nordic-hid/fu-nordic-hid-cfg-channel.c
Comment (hughsie) [inline plugins/nordic-hid/fu-nordic-hid-cfg-channel.c:697]: `too many sync attempts` perhaps -- and I do wonder if fu_device_retry_full might be better to use here.

File: plugins/nordic-hid/fu-nordic-hid-archive.c
Comment (d4s) [reply plugins/nordic-hid/fu-nordic-hid-archive.c:26]: Thanks!!!

File: plugins/nordic-hid/fu-nordic-hid-cfg-channel.c
Comment (d4s) [reply plugins/nordic-hid/fu-nordic-hid-cfg-channel.c]: ok, reworked a bit. However crc32 might be zero, afair, so have no sense to check it against 0.
Instead I set the '-1' for requested checksum kind, hence checksum will work only for the custom calculation function and added error handling to firmware's implementation.

File: plugins/nordic-hid/fu-nordic-hid-cfg-channel.c
Comment (d4s) [reply plugins/nordic-hid/fu-nordic-hid-cfg-channel.c:152]: @hughsie do you think we need to support Bluez versions <5.56?

Comment (hughsie) [reply plugins/nordic-hid/fu-nordic-hid-cfg-channel.c:152]: Upstream maintainer hat on, "no", but it depends on whether nordic wants this backported to earlier versions. Personally, I'd KISS.

Comment (d4s) [reply plugins/nordic-hid/fu-nordic-hid-cfg-channel.c:152]: Ok, I'll leave it as is for now then. In case if we need to support old version -- we know the place to fix.

**Comment by d4s** [comment]
This is the initial version supporting/tested with Nordic nrf52840dk board.
- configuration channel is used for updates
- B0 bootloader type is tested

Will squash before merge

**Comment by hughsie** [comment]
I think rebase against master and squash into one commit. Then I'll do one more round.

**Comment by hughsie** [comment]
@superm1 this LGTM, but can you do a quick pass with fresh eyes please?

**Review by hughsie — APPROVED** [review]

**Review by superm1 — APPROVED** [review]

**Pull request merged**
